src/mongroup.cpp

```diff
@@ -297,6 +293,26 @@ void MonsterGroupManager::FinalizeMonsterGroups()
             debugmsg( "monster on blacklist %s does not exist", mtid.c_str() );
         }
     }
+    // If we have the classic zombies option, remove non-conforming monsters
+    if ( get_option<bool>( "CLASSIC_ZOMBIES" ) ) {
+        for( auto &elem : monsterGroupMap ) {
+            MonsterGroup &mg = elem.second;
+            for( FreqDef::iterator c = mg.monsters.begin(); c != mg.monsters.end(); ) {
+                // Test mon
+                const mtype &mt = c->name.obj();
+                
+                bool in_classic = mt.in_category( "CLASSIC" );
+                bool in_wildlife= mt.in_category( "WILDLIFE" );
+
+                if ( !( in_classic | in_wildlife ) ){
```

Comment:
Maybe you meant `||`?

Reply:
You are right. Additionally, after looking through the generated assembly it's worth more to not precompute the values `in_classic` and `in_wildlife`, instead inlining their computation to the conditional. Will be updating shortly.